sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Testing/EventHubScope.cs

```diff
@@ -104,6 +139,44 @@ public async ValueTask DisposeAsync()
         public static Task<EventHubScope> CreateAsync(int partitionCount,
                                                       [CallerMemberName] string caller = "") => CreateAsync(partitionCount, Enumerable.Empty<string>(), caller);
 
+        /// <summary>
+        ///   Performs the tasks needed to get or create a new Event Hub instance with the requested
+        ///   partition count and a dynamically assigned unique name.
+        /// </summary>
+        ///
+        /// <param name="partitionCount">The number of partitions that the Event Hub should be configured with.</param>
+        /// <param name="consumerGroups">The set of consumer groups to create and associate with the Event Hub; the default consumer group should not be included, as it is implicitly created.</param>
+        /// <param name="caller">The name of the calling method; this is intended to be populated by the runtime.</param>
+        ///
+        public static Task<EventHubScope> CreateAsync(int partitionCount,
+                                                      IEnumerable<string> consumerGroups,
+                                                      [CallerMemberName] string caller = "") => CreateIfMissingAsync(partitionCount, consumerGroups, caller);
+
+        /// <summary>
+        ///   It returns the eventhub that is named using <see cref="TestEnvironment.EventHubName" />.
+        ///   It creates and returns one otherwise.
+        ///
+        ///   If a hub was created, it would be disposed afterwards.
+        /// </summary>
+        ///
+        /// <param name="partitionCount">The number of partitions that the Event Hub should be configured with.</param>
+        /// <param name="consumerGroups">The set of consumer groups to create and associate with the Event Hub; the default consumer group should not be included, as it is implicitly created.</param>
+        /// <param name="caller">The name of the calling method; this is intended to be populated by the runtime.</param>
+        ///
+        /// <returns>The <see cref="EventHubScope" /> that will be used in a given test run.</returns>
+        ///
+        private static Task<EventHubScope> CreateIfMissingAsync(int partitionCount,
```

Comment:
nit:  Sorry that I didn't catch this earlier, but can we rename this to something like `BuildScope` or similar?    I found myself getting a bit confused between `CreateAsync` which callers use to create the scope and some of the `private` members that are making decisions around individual resources used by the scope.

This method is making the decision on how to build up the scope - either using the "we already have resources" path or the "we need to create resources" path.

Reply:
In ```StorageTestEnvironment``` we used ```EnsureStorageAccount``` to perform something similar:

```csharp
private static StorageScope.StorageProperties EnsureStorageAccount()
{
    if (!string.IsNullOrEmpty(ActiveStorageAccountConnectionString.Value))
    {
        return StorageScope.PopulateStoragePropertiesFromConnectionString(ActiveStorageAccountConnectionString.Value);
    }

    return CreateStorageAccount();
}
```

Reply:
I like the `Ensure` naming.